saleor/graphql/checkout/mutations/checkout_create.py

```diff
@@ -376,7 +411,35 @@ def save(cls, info: ResolveInfo, instance: models.Checkout, cleaned_input):
                 instance.billing_address = billing_address.get_copy()
 
             instance.save()
-            create_checkout_metadata(instance)
+
+    @classmethod
+    def post_save_action(cls, info: ResolveInfo, instance, cleaned_input):
+        # Write metadata in post-save action because in "save" metadata doesn't exist in
+        # cleaned_input.
+
+        checkout_metadata = create_checkout_metadata(instance)
+
+        metadata_from_input = cleaned_input.get("metadata", [])
+        private_metadata_from_input = cleaned_input.get("private_metadata", [])
+
+        nothing_to_write = (
+            len(metadata_from_input) == 0 and len(private_metadata_from_input) == 0
+        )
+
+        if nothing_to_write:
+            return
+
+        metadata_dict = {
```

Comment:
Why do you need to go through that metadata if it's already in key, value pairs? If there is a reason maybe you can add a comment about it?

Reply:
> Why do you need to go through that metadata if it's already in key, value pairs?

Input is a list of objects: [ {key:"key provided by user", "value": "value provided by user}]